linode/manager
# chore: [M3-6496] - Add Adobe Analytics custom event tracking

## Description 📝

**What does this PR do?**
This PR enables custom event tracking in Adobe Analytics using a Direct Call rule and the Adobe [`_satellite`](https://experienceleague.adobe.com/docs/experience-platform/tags/client-side/satellite-object.html?lang=en) object. 

Everything but the changes in `packages/manager/src/utilities/ga.ts` will be merged before this PR in step 1: #8989, which sets up basic Adobe page view tracking on Cloud Manager. Step 3, removal of Google Analytics, will come in a later PR.

This PR is marked as Do Not Merge for now to roll out the analytics migration in phases. It can still be reviewed but should not be merged until **after** 4/26 (date next release branch is cut) and is scheduled for the **5/15** release. I will update this PR if this timeline changes.

## How to test 🧪

**What are the steps to reproduce the issue or verify the changes?**
1.  Check out this PR locally.
2. Test that there are no unexpected errors when running Cloud Manager prior to defining an Adobe Analytics env var for local testing.
3. Set the `REACT_APP_ADOBE_ANALYTICS_URL` environment variable in `.env`. Reach out if you don't have this.
4. Open the browser tools Network tab, filter requests by "adobe", and check that successful network requests have been made to load the launch script.
5. In the browser console, type `_satellite.setDebug(true)`. 
6. Refresh the page. You should see something like the following, indicating that Adobe Analytics is integrated into the website and is performing basic tracking of page views:
![image](https://user-images.githubusercontent.com/114685994/232653431-ca50fa95-54d0-4cb2-aff7-48bae3245030.png)
7. Test out a handful of custom events defined in `packages/manager/src/utilities/ga.ts`. After triggering each event, check that the Adobe Analytics debug log in the browser console has fired a corresponding direct call custom event with the additional details (category, action, label [if applicable], and value [if applicable]). For example, `sendCreateVolumeEvent` triggers the following direct call rule when the `sendEvent` function is called on "Create Volume" button click:
![image](https://user-images.githubusercontent.com/114685994/232662712-c0acdc1f-fa2f-42a0-8a33-d9663284a5f9.png)
9. Verify that the E2E tests pass without any unexpected error.

**How do I run relevant unit or e2e tests?**
``` 
yarn cy:run
```
